checkmate/async/celery.py

```diff
@@ -47,7 +52,16 @@ def bootstrap_worker(sender, **_kwargs):  # pragma: no cover
     """Set up the celery worker with one-time initialisation."""
 
     # Put some common handy things around for tasks
-    checkmate = create_app(celery_worker=True)
+    try:
+        checkmate = create_app(celery_worker=True)
+
+    # pylint: disable=broad-except
+    except Exception as err:
+        # If we don't bail out here ourselves, Celery just hides the error
+        # and continues. This means `request_context` will not be available
+        # when attempting to run tasks, which is impossible to debug.
+        LOG.fatal("CELERY WORKER DID NOT START: Could not create app: %s", err)
```

Comment:
`logging.fatal()` is [undocumented](https://docs.python.org/3.6/library/logging.html) (it's an alias for `logging.critical()` which is documented).

Passing `err` to `logging.whatever()` as part of the string formatting just logs the `str()` representation of the exception object, it doesn't properly log the exception with traceback and everything.

I think there are two right ways to log an exception from an exception handler, and you don't have to pass the exception object to `logging` yourself:

1. Just call `logging.exception()`, although this logs with level `error` not `critical`

   ```python
   try:
       raise RuntimeError("Halp")
   except Exception:
       LOG.exception("Error happened")
   ```

2. Call `logging.critical()` (or any other logging method) and pass `exc_info=True`, that will log an exception but with the logging level that you want:

   ```python
   try:
       raise RuntimeError("Halp")
   except Exception:
       LOG.critical("Error happened", exc_info=True)
   ```

(You can also pass an exception object or `exc_info` tuple as the`exc_info` argument instead of `True` if you want to log a particular exception. But if you're in an `except` block you don't have to, it just uses `sys.exc_info()` to find the current exception.)

Reply:
I assume the `fatal` redirect is for old fogies like me that haven't noticed it's been changed. It's been `fatal` since the Log4J days (https://www.tutorialspoint.com/log4j/log4j_logging_levels.htm). So I guess they diverged at some point in Python.

I've added the `exc_info`. Didn't know about that; very handy.